### PR TITLE
Add example test case

### PR DESCRIPTION
**Pull request by Xammie**
I think adding a working example test case would improve the onboarding experience. I'm not sure if I have used the `IntegrationTest` class correctly so please let me know if I need to change anything. 

Also think there could be a documentation page about testing, would you accept a PR for it?

### CODE REVIEW

File: tests/HomeControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+declare(strict_types=1);
+
+namespace Tests;
+
+use Tempest\Framework\Testing\IntegrationTest;
+
+class HomeControllerTest extends IntegrationTest
+{
+    protected string $root = __DIR__ . '/../';
```
Comment (Xammie) [inline tests/HomeControllerTest.php:11]: This is needed in order for the TestCase to find `composer.json`. Would be nice if this was not required

**Comment by brendt** [comment]
> Also think there could be a documentation page about testing, would you accept a PR for it?

Yes, for sure!

File: tests/HomeControllerTest.php (hunk 1 above)
Comment (brendt) [reply tests/HomeControllerTest.php:11]: Hmm yeah, the way I work around that in `tempest/framework` is by implementing an abstract `FrameworkIntegrationTestCase`, which all other tests extend from. That in turn extends from `IntegrationTest`. 

I'm not opposed to having a `ProjectIntegrationTestCase` as well, shipped with `tempest/app`. WDYT?

For reference: https://github.com/tempestphp/tempest-framework/blob/main/tests/Integration/FrameworkIntegrationTestCase.php

**Comment by brendt** [comment]
Nice PR, btw! Let's discuss the `ProjectTestCase` idea, and see what's next before merging :)

File: tests/HomeControllerTest.php (hunk 1 above)
Comment (Xammie) [reply tests/HomeControllerTest.php]: I think its a good idea to ship an `IntegrationTestCase`. I'm not sure about what would be the best name for it. I've used `IntegrationTestCase` for now, but open to other suggestions.

**Comment by Xammie** [comment]
I've written a draft for the documentation https://github.com/tempestphp/tempest-app/pull/8. 

Wondering if there is anything I can add or change for this MR? :)

**Comment by brendt** [comment]
Let's merge!

**Pull request merged**